File: src/main/java/com/dinaraparanid/converter/TrackFileExtension.java
package com.dinaraparanid.converter;

import org.jetbrains.annotations.NotNull;

import java.util.Map;

public enum TrackFileExtension {
    MP3 {
        @Override
        public String getExtension() {
            return "mp3";
        }
    },

    MP4 {
        @Override
        public String getExtension() {
            return "mp4";
        }
    },

    WAV {
        @Override
        public String getExtension() {
            return "wav";
        }
    },

    AAC {
        @Override
        public String getExtension() {
            return "aac";
        }
    },

    FLAC {
        @Override
        public String getExtension() {
            return "flac";
        }
    },

    M4A {
        @Override
        public String getExtension() {
            return "m4a";
        }
    },

    OPUS {
        @Override
        public String getExtension() {
            return "opus";
        }
    },

    VORBIS {
        @Override
        public String getExtension() {
            return "vorbis";
        }
    };

    @NotNull
    private static final Map<String, TrackFileExtension> extensionMap = TrackFileExtensionExtKt.getExtensionMap();

    @NotNull
    public static TrackFileExtension fromString(@NotNull String extension) {
        return extensionMap.get(extension);
    }

    @NotNull
    public abstract String getExtension();

    @Override
    public String toString() {
        return getExtension();
    }
}

File: src/main/kotlin/com/paranid5/vtmserver/domain/TrackFileExtension.kt
package com.paranid5.vtmserver.domain

enum class TrackFileExtension {
    MP3, WAV, AAC, FLAC, M4A, OPUS, VORBIS;

    val extension get() = toString().lowercase()

    companion object {
        private val extensionMap by lazy {
            hashMapOf(
                *TrackFileExtension.values()
                    .map { it.extension to it }
                    .toTypedArray()
            )
        }

        fun fromString(extension: String) = extensionMap[extension]!!
    }
}